src/splice.ts

```diff
@@ -15,19 +136,29 @@ import { purry } from "./purry";
  * @dataFirst
  * @category Array
  */
-export function splice<T>(
-  items: ReadonlyArray<T>,
-  start: number,
-  deleteCount: number,
-  replacement: ReadonlyArray<T>,
-): Array<T>;
+export function splice<
+  T extends IterableContainer,
+  Start extends number,
+  DeleteCount extends number,
+  Replacement extends Readonly<WidenTuple<T>>,
```

Comment:
Wouldn't this typing cause typo's in arrays of enum (literal union) values break silently, e.g.

```ts
type MyEnum = "cat" | "dog";

const DATA = ["cat"] as const satisfies readonly MyEnum[];

splice(DATA, 0, 0, ["bat"]);
```